### DISCUSSION TITLE
Easy integration with COSMOS

### DISCUSSION THREAD

**Discussion by davidskeck** [nasa/openmct#3846] [Community Q&A] [answered]
[COSMOS](https://github.com/BallAerospace/COSMOS) is a well-used tool and it would be great to have out-of-the-box integration with OpenMCT. Is there something OpenMCT community could do to support this?

**Comment by akhenry** [comment]
We would love for someone to do an open source plugin for integrating with COSMOS. This would be of huge value to a lot of potential users. If someone wanted to attempt it we would be happy to provide technical support and guidance.

**Comment by davidskeck** [comment]
That's great to hear. Can you tell me if the latest API for OpenMCT has tutorials as of yet? It looks like the OpenMCT tutorial repository hasn't been updated for 2 years.

**Comment by imrishabh18** [comment]
Hey! @akhenry I would like to contribute to this repo and will really appreciate your guidance. Is this a beginner friendly issue to start with?

**Reply by akhenry** [reply]
I do actually think it's pretty beginner friendly for someone with a good grasp of JavaScript, but who is new to Open MCT. Everything you need to know about Open MCT to address this issue is covered in our [telemetry integration tutorial](https://github.com/nasa/openmct-tutorial). The [Open MCT-YAMCS plugin](https://github.com/akhenry/openmct-yamcs) is also a useful reference. YAMCS is another open source telemetry and commanding system, like COSMOS. 

It would require a decent investment of time, but it would be a fun project that would probably garner a decent amount of interest. There are definitely missions using COSMOS who would be interested in Open MCT if that integration was available. There is also a corresponding issue in the COSMOS repository - https://github.com/BallAerospace/COSMOS/issues/935

BTW, a COSMOS plugin should be created in a new repository (ie. not as a PR to the Open MCT repository).

**Comment by jvigliotta** [comment]
Hello! I'm going to actually move this over to discussions as I think it's a good topic for the Community Q&A

**Comment by davidskeck** [comment, marked as answer]
COSMOS is moving to a web view in v5.0 so I think we can go ahead and close this. It will cover my use case.
